client/me/purchases/manage-purchase/index.tsx

```diff
@@ -639,6 +642,76 @@ class ManagePurchase extends Component<
 		return null;
 	}
 
+	renderRefundText() {
+		const { purchase, translate } = this.props;
+
+		// Hide if refund window has lapsed.
+		if ( ! purchase?.isRefundable || ! purchase?.refundText || ! purchase?.mostRecentRenewDate ) {
+			return;
+		}
+
+		const ONE_DAY_IN_MILLISECONDS = 24 * 60 * 60 * 1000;
+		const lastRenewalDate = new Date( purchase.mostRecentRenewDate );
+		const refundPeriodEndDate = new Date(
+			lastRenewalDate.getTime() + purchase.refundPeriodInDays * ONE_DAY_IN_MILLISECONDS
+		);
+
+		const refundText = (() => {
+			// User would receive another free domain credit upon cancelling.
+			if ( isDomainRegistration( purchase ) && ! hasAmountAvailableToRefund( purchase ) ) {
+			  return translate(
+				"If you're unsure about this domain, cancel it before {{refundDate/}} to choose another free domain instead.",
```

Comment:
I don't believe this text is accurate.  Domain credits are one-time use; you can't use them over and over again.

Reply:
Sorry - silly question! - would you mind clarifying whether “believe” here means they’re definitely not refundable or would it be better to check with someone else? I haven’t tested since I don’t have a free domain credit on my account, but I asked my HE Advisor this week who said that “cancelling the domain should open up a new domain credit”.

Reply:
I'd have to search the refund code but I'm pretty sure David is right. If you've used a domain credit, I don't think there is code anywhere that restores it just because you've cancelled the domain within the refund period.

Reply:
Interesting! That seems like it would be useful, but I've removed that line from the code here for now.

Reply:
Yes, I tested it now and confirmed that the domain credit isn't restored after the domain is cancelled.  You could point your HE Advisor to this internal documentation which explains it in more detail: P6Ya1B-xS-p2

I thought there was public documentation about this too, but perhaps not... That may be because it's not necessarily a hard and fast rule (at least when HEs are involved), but in terms of what customers are allowed to do on their **own** in Calypso, it's intentional that they are restricted from using the domain credit over and over again.

Reply:
Yes, from my experience, the domain credit is always returned when people ask - but also makes sense that they might need to go through that process to prevent abuse etc.